pkgs/development/tools/continuous-integration/gitlab-runner/default.nix

```diff
@@ -0,0 +1,64 @@
+{ lib, buildGoPackage, fetchFromGitLab, fetchurl, go-bindata }:
+
+buildGoPackage rec {
+  version = "1.5.3";
+  name = "gitlab-runner-${version}";
+  goPackagePath = "gitlab.com/gitlab-org/gitlab-ci-multi-runner";
+  commonPackagePath = "${goPackagePath}/common";
+  buildFlagsArray = ''
+    -ldflags=
+      -X ${commonPackagePath}.NAME=gitlab-runner
+      -X ${commonPackagePath}.VERSION=${version}
+      -X ${commonPackagePath}.REVISION=v${version}
+  '';
+
+  src = fetchFromGitLab {
+    owner = "gitlab-org";
+    repo = "gitlab-ci-multi-runner";
+    rev = "v${version}";
+    sha256 = "1rs70jcgi1x9vpfbx4fay0j025amvrj48wgfwk7vwzy9rbdizka4";
+  };
+
+  # Gitlab runner embeds some docker images these are prebuilt for arm and x86_64
+  docker_x86_64 = fetchurl {
+    url = "https://gitlab-ci-multi-runner-downloads.s3.amazonaws.com/v${version}/docker/prebuilt-x86_64.tar.xz";
+    sha256 = "108cszdn9616q5ls86hfcjvgh6cp8jb3g4h94c9ag63f0z8zxzx2";
+  };
+
+  docker_arm = fetchurl {
+    url = "https://gitlab-ci-multi-runner-downloads.s3.amazonaws.com/v${version}/docker/prebuilt-arm.tar.xz";
+    sha256 = "0axiq9yxkybrdg0h59wr6jm2xq3aw8dwv7mqfk1a6k2j8bxk9z08";
+  };
```

Comment:
I tried that, but I require `version` for the download the correct version. Should i move version into the let part too?
Anyway I still don't 100% understand what the advantage is?